paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java

```diff
@@ -217,18 +217,49 @@ public RecordReader<InternalRow> createReader(Split split) {
                 throw new IllegalArgumentException("Unsupported split: " + split.getClass());
             }
             Path location = ((TagsSplit) split).location;
-            LeafPredicate predicate = ((TagsSplit) split).tagName;
+            Predicate predicate = ((TagsSplit) split).tagPredicate;
             TagManager tagManager = new TagManager(fileIO, location, branch);
+            String leafName = "tag_name";
```

Comment:
first : private static final TAG_NAME = "tag_name";
second : replace all "tag_name" to TAG_NAME in this class